web/backend/src/controllers/dela.ts

```diff
@@ -0,0 +1,253 @@
+    payload = getPayload(dataStr2);
+    redirectToDefaultProxy = false;
+  }
+
+  // in case this is a DKG setup request, we must update the payload.
+  const dkgSetupRegex = /\/evoting\/services\/dkg\/actors\/.*$/;
+  if (uri.match(dkgSetupRegex)) {
+    const dataStr2 = JSON.stringify({ Action: req.body.Action });
+    payload = getPayload(dataStr2);
+
+    // If setup don't redirect to default proxy, if 'computePubshares' then keep
+    // default proxy
+    if (req.body.Action === 'setup') {
+      redirectToDefaultProxy = false;
+    }
+  }
+
+  // in case this is a DKG init or setup request, we must extract the proxy addr
+  if (!redirectToDefaultProxy) {
+    const proxy = req.body.Proxy;
+
+    if (proxy === undefined) {
+      res.status(400).send('proxy undefined in body');
+      return;
+    }
+    uri = proxy + req.baseUrl.slice(4);
+  }
+
+  console.log('sending payload:', JSON.stringify(payload), 'to', uri);
+
+  axios({
+    method: req.method as Method,
+    url: uri,
+    data: payload,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  })
+    .then((resp) => {
+      res.status(200).send(resp.data);
+    })
+    .catch((error: AxiosError) => {
+      let resp = '';
+      if (error.response) {
+        resp = JSON.stringify(error.response.data);
+      }
+      console.log(error);
+
+      res
+        .status(500)
+        .send(`failed to proxy request: ${req.method} ${uri} - ${error.message} - ${resp}`);
+    });
+}
+
+// Secure /api/evoting to admins and operators
+delaRouter.put('/authorizations', (req, res) => {
+  if (!req.session.userId) {
+    res.status(400).send('Unauthorized');
+    return;
+  }
+  if (
+    !isAuthorized(req.session.userId, PERMISSIONS.SUBJECTS.ELECTION, PERMISSIONS.ACTIONS.CREATE)
+  ) {
+    res.status(400).send('Unauthorized');
+    return;
+  }
+  const { FormID } = req.body;
+  assignUserPermissionToOwnElection(String(req.session.userId), FormID);
+});
+
+// https://stackoverflow.com/a/1349426
+function makeid(length: number) {
+  let result = '';
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  const charactersLength = characters.length;
+  for (let i = 0; i < length; i += 1) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength));
+  }
+  return result;
+}
+delaRouter.put('/forms/:formID', (req, res, next) => {
+  const { formID } = req.params;
+  if (!isAuthorized(req.session.userId, formID, PERMISSIONS.ACTIONS.OWN)) {
+    res.status(400).send('Unauthorized');
+    return;
+  }
+  next();
+});
+
+delaRouter.post('/services/dkg/actors', (req, res, next) => {
+  const { FormID } = req.body;
+  if (!isAuthorized(req.session.userId, FormID, PERMISSIONS.ACTIONS.OWN)) {
+    res.status(400).send('Unauthorized');
+    return;
+  }
+  if (FormID === undefined) {
+    return;
+  }
+  next();
+});
+delaRouter.use('/services/dkg/actors/:formID', (req, res, next) => {
```

Comment:
I know it's not your fault, but while you're at it, can we add newlines in between all these endpoints ? this hurts my eyes 😅

Reply:
No problem at all..  I already added them in the following PR https://github.com/dedis/d-voting/pull/349